### PR TITLE
feat: 1694 lock ownership to read only

### PR DESCRIPTION
**Pull request by xiaopeng0202**
This pull request introduces a new API endpoint to retrieve seedlot information by seedlot number and integrates this functionality into the frontend. The changes include updates to the API configuration, addition of a new endpoint in the backend, and modifications to the frontend components to utilize this new endpoint.

New
- lock ownership to read only when original qty is over 0
- added new endpoint on oracle db to expose original qty from seedlot table

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [ ] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-12-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1712-backend.apps.silver.devops.gov.bc.ca/swagger-ui/index.html)
- [Oracle-API](https://nr-spar-1712-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

### CODE REVIEW

**Comment by DerekRoberts** [comment]
@xiaopeng0202 Any sense writing a test to ensure write only?

File: frontend/src/api-service/ApiConfig.ts [hunk 1]
```diff
@@ -55,7 +55,9 @@ const ApiConfig = {
 
   areaOfUseSpzList: `${oracleServerHost}/api/area-of-use/spz-list/vegetation-code`,
 
-  parentTreeByVegCode: `${oracleServerHost}/api/parent-trees/vegetation-codes/{vegCode}`
+  parentTreeByVegCode: `${oracleServerHost}/api/parent-trees/vegetation-codes/{vegCode}`,
+
+  seedlotBySeedlotNumber: `${oracleServerHost}/api/seedlot/{seedlotNumber}`
```
Comment (mgaseta) [inline frontend/src/api-service/ApiConfig.ts:60]: Maybe we should rename this to something like "getSeedlotFromOrableDb" or something like that, to better differentiate this path.

File: frontend/src/api-service/seedlotAPI.ts [hunk 2]
```diff
@@ -72,3 +72,8 @@ export const getAClassSeedlotFullForm = (seedlotNumber: string) => {
   const url = `${ApiConfig.seedlots}/${seedlotNumber}/a-class-full-form`;
   return api.get(url).then((res) => res.data as SeedlotAClassFullResponseType);
 };
+
+export const getSeedlotBySeedlotNumber = (seedlotNumber: string) => {
```
Comment (mgaseta) [inline frontend/src/api-service/seedlotAPI.ts:76]: Same here, a different name to know this is coming from oracle.

File: frontend/src/components/SeedlotRegistrationSteps/OwnershipStep/index.tsx [hunk 3]
```diff
@@ -143,6 +151,9 @@ const OwnershipStep = ({ isReview }: OwnershipStepProps) => {
 
   const getFcQuery = (clientNumber: string): ForestClientType | undefined => qc.getQueryData(['forest-clients', clientNumber]);
 
+  const originalSeedQty = getSeedlotBySeedlotNumberQuery.data?.data?.originalSeedQty ?? 0;
+  const readOnly = originalSeedQty > 0;
```
Comment (mgaseta) [inline frontend/src/components/SeedlotRegistrationSteps/OwnershipStep/index.tsx:155]: I think this const can have a less generic name.

File: oracle-api/config/application-local.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+spring:
+  datasource:
+    url: jdbc:oracle:thin:@tcp://localhost:1521/dbdock_01
+    username: THE
+    password: default
```
Comment (mgaseta) [inline oracle-api/config/application-local.yml:5]: Question: what is this? (And maybe remove one of the empty lines?)

**Review by mgaseta — CHANGES_REQUESTED** [review]
Nice job!
Added only a few markups, but nothing major.

File: oracle-api/config/application-local.yml (hunk 4 above)
Comment (xiaopeng0202) [reply oracle-api/config/application-local.yml:5]: it's the config to running oracle db locally

File: frontend/src/views/Seedlot/SeedlotReview/SeedlotReviewContent.tsx [hunk 5]
```diff
@@ -550,24 +550,15 @@ const SeedlotReviewContent = () => {
...
-              renderIcon={isReadMode ? Edit : Save}
-              onClick={handleEditSaveBtn}
-              disabled={getSeedlotBySeedlotNumberQuery.isSuccess
-                && getSeedlotBySeedlotNumberQuery.data.data.originalSeedQty > 0}
-            >
-              {isReadMode ? 'Edit seedlot' : 'Save edit'}
-            </Button>
-          )
-          : null
-      }
```
Comment (mgaseta) [inline frontend/src/views/Seedlot/SeedlotReview/SeedlotReviewContent.tsx]: Shouldn't we keep this for historic seedlots that are CUS or UPT?

File: frontend/src/views/Seedlot/SeedlotReview/SeedlotReviewContent.tsx [hunk 6]
```diff
@@ -107,7 +107,7 @@ const SeedlotReviewContent = () => {
 
   const getSeedlotBySeedlotNumberQuery = useQuery({
     queryKey: ['seedlot-by-seedlotNumber', seedlotNumber],
-    queryFn: () => getSeedlotBySeedlotNumber(seedlotNumber ?? ''),
+    queryFn: () => getSeedlotFromOracleDbBySeedlotNumber(seedlotNumber ?? ''),
     enabled: seedlotQuery.isFetched
   });
```
[the remark this answers is not in the document]
Comment (xiaopeng0202) [reply frontend/src/views/Seedlot/SeedlotReview/SeedlotReviewContent.tsx]: sorry for that, it was a merge conflit, just made update

File: frontend/src/views/Seedlot/SeedlotReview/SeedlotReviewContent.tsx (hunk 5 above)
Comment (xiaopeng0202) [reply frontend/src/views/Seedlot/SeedlotReview/SeedlotReviewContent.tsx]: https://github.com/bcgov/nr-spar/pull/1721

We have another PR to remove it

**Review by mgaseta — APPROVED** [review]
LGTM!

**Pull request merged**